### PR TITLE
RET-4528

### PR DESCRIPTION
**Pull request by cindychmcts**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/RET-4528

### Change description ###

Auto populate any reasonable adjustments submitted on the ET1 into the relevant section in the ET1 vetting event

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java [hunk 1]
```diff
@@ -89,6 +92,21 @@ public void initialiseEt1Vetting(CaseDetails caseDetails) {
...
+        if (!ObjectUtils.isEmpty(caseData.getClaimantHearingPreference())
+            && !StringUtils.isEmpty(caseData.getClaimantHearingPreference().getReasonableAdjustments())) {
+            if (YES.equals(caseData.getClaimantHearingPreference().getReasonableAdjustments())) {
+                caseData.setEt1ReasonableAdjustmentsQuestion(YES);
+            } else {
+                caseData.setEt1ReasonableAdjustmentsQuestion(NO);
+            }
+            caseData.setEt1ReasonableAdjustmentsTextArea(
+                caseData.getClaimantHearingPreference().getReasonableAdjustmentsDetail());
+        }
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java:109]: Could we use guard clauses here instead?

also `caseData.getClaimantHearingPreference()` and its `getReasonableAdjustments()` is called many times, might be better to extract as a variable.

**Review by JackReevies — CHANGES_REQUESTED** [review]

**Review by btensay — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java [hunk 2]
```diff
@@ -89,6 +93,28 @@ public void initialiseEt1Vetting(CaseDetails caseDetails) {
...
+            return;
+        }
+
+        if (YES.equals(reasonableAdjustments)) {
+            caseData.setEt1ReasonableAdjustmentsQuestion(YES);
+        } else {
+            caseData.setEt1ReasonableAdjustmentsQuestion(NO);
+        }
+
+        caseData.setEt1ReasonableAdjustmentsTextArea(hearingPreference.getReasonableAdjustmentsDetail());
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java:117]:
```suggestion
        Optional.ofNullable(hearingPreference)
                .map(ClaimantHearingPreference::getReasonableAdjustments)
                .ifPresent(reasonableAdjustment -> {
                    caseData.setEt1ReasonableAdjustmentsQuestion(YES.equals(reasonableAdjustment) ? YES : NO);
                    caseData.setEt1ReasonableAdjustmentsTextArea(hearingPreference.getReasonableAdjustmentsDetail());
                });
```

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java [hunk 3]
```diff
@@ -4,7 +4,9 @@
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java:7]:
```suggestion
```

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java [hunk 4]
```diff
@@ -4,7 +4,9 @@
...
 import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et1VettingService.java:9]:
```suggestion
```

**Review by JackReevies — CHANGES_REQUESTED** [review]

**Review by JackReevies — APPROVED** [review]

**Review by jip1984 — APPROVED** [review]

**Pull request merged**